addons/edi/models/edi_synchronizer.py

```diff
@@ -263,7 +270,10 @@ def elide(self, doc, vlist):
 
                     # Elide EDI records that would not change the target record
                     target_vals = self.target_values(record_vals)
-                    if all(comparator[k](target[k], v) for k, v in target_vals.items()):
+                    changed_vals = {k: v for k, v in target_vals.items() if not comparator[k](target[k], v)}
+                    if changed_vals:
+                        record_vals["changed_fields"] = str(list(changed_vals.keys()))
```

Comment:
Since I suggest to split by `,` in my other comment, this probably should be a `",".join(...)` or similar.